src/main/java/org/veupathdb/service/eda/compute/plugins/correlation/CorrelationPlugin.java

```diff
@@ -0,0 +1,332 @@
+        connection.voidEval(util.getVoidEvalFreadCommand(INPUT_DATA, metadataInputVars));
+        connection.voidEval("sampleMetadata <- " + INPUT_DATA); 
+
+         // Prep data and run correlation
+        if (isEigengene)  {
+          // If we have eigenegene data, we'll use our base correlation function in veupathUtils, so we
+          // only need to make data frames for the assay data and sample metadata.
+          connection.voidEval("eigengeneData <- assayData[order(" + computeEntityIdColName + ")]; " + 
+            "eigengeneData <- eigengeneData[, -as.character(" + dotNotatedEntityIdColumnsString + "), with=FALSE];" +
+            "eigengeneData <- eigengeneData[, -" + singleQuote(computeEntityIdColName) + ", with=FALSE];");
+
+          connection.voidEval("sampleMetadata <- sampleMetadata[order(" + computeEntityIdColName + ")]; " + 
+            "sampleMetadata <- sampleMetadata[, -as.character(" + dotNotatedEntityIdColumnsString + "), with=FALSE];" +
+            "sampleMetadata <- sampleMetadata[, -" + singleQuote(computeEntityIdColName) + ", with=FALSE];");
+        
+          connection.voidEval("computeResult <- veupathUtils::correlation(data1=eigengeneData, data2=sampleMetadata" +
+                                    ", method=" + singleQuote(method) +
+                                    ", verbose=TRUE)");
+        } else {
+          
+          // If we don't have eigengene data, for now we can assume the data is abundance data.
+          // Abundance data can go through our microbiomeComputations pipeline.
+          connection.voidEval("sampleMetadata <- microbiomeData::SampleMetadata(data = sampleMetadata" +
+                                    ", recordIdColumn=" + singleQuote(computeEntityIdColName) +
+                                    ", ancestorIdColumns=as.character(" + dotNotatedEntityIdColumnsString + "))");
+
+          connection.voidEval("abundanceData <- microbiomeData::AbundanceData(data=assayData" + 
+                                    ", sampleMetadata=sampleMetadata" +
+                                    ", recordIdColumn=" + singleQuote(computeEntityIdColName) +
+                                    ", ancestorIdColumns=as.character(" + dotNotatedEntityIdColumnsString + ")" +
+                                    ", imputeZero=TRUE)");       
+                                  
+          // Run correlation!
+          connection.voidEval("computeResult <- veupathUtils::correlation(data1=abundanceData" +
+                                                              ", method=" + singleQuote(method) +
+                                                              proportionNonZeroThresholdRParam +
+                                                              varianceThresholdRParam +
+                                                              stdDevThresholdRParam +
+                                                              ", verbose=TRUE)");
+        }
+      } else {
```

Comment:
A comment or two here to help us know where we are in the conditional would be helpful please!

Reply:
i was also considering that there should be a helper method somewhere something like `makeCollectionsDataRObject` that takes the collectionSpec and data handle, optionally two of each of those, and builds whatever is appropriate for the member type its provided. thatd probably make the flow clearer, and make it easier to grow into additional data types.

Reply:
thinking about it a bit.. im gonna add some clarifying comments for now. and when i do the prefilters, i think i might end up introducing a new class for the eigengenes too. if that happens, ill add a helper like i talked about to edacommon then probably.